test/test_doc_legacy_convert.py

```diff
@@ -13,3 +21,20 @@ def test_new_to_old():
     doc = DoclingDocument.model_validate(dict_from_yaml)
 
     docling_document_to_legacy(doc)
+
+
+def test_old_to_new():
+    filepath = Path("test/data/legacy_doc/doc-export.json")
+    leg_doc = DsDocument.model_validate_json(filepath.read_text())
+
+    doc = legacy_to_docling_document(leg_doc)
+
```

Comment:
Can you add here the 

```python
 leg_doc2 = docling_document_to_legacy(doc)
 assert leg_doc==leg_doc2
 ```

Reply:
Changes found:
- name of labels (upper vs lower case)
- caption of figures are not in main-text anymore
- s3_data removed
- model metadata removed
- logs removed
- document hash cannot be preserved